src/c++/library/grpc_client.cc

```diff
@@ -691,6 +691,87 @@ InferenceServerGrpcClient::ModelInferenceStatistics(
   return err;
 }
 
+Error
+InferenceServerGrpcClient::UpdateTraceSettings(
+    const std::string& model_name,
+    const std::map<std::string, std::vector<std::string>>& settings,
+    const Headers& headers)
+{
+  Error err;
+
+  inference::TraceSettingRequest request;
+  inference::TraceSettingResponse response;
```

Comment:
Shouldn't the update API also returns the response (server will return the updated trace setting)?

Reply:
Please refer to `InferenceServerGrpcClient::ModelInferenceStatistics()` for how to return the response. We still want to return error in the case of failure instead of logging it, so that the user can detect and handle error programmatically.